api/rosetta/errors.go

```diff
@@ -193,3 +204,69 @@ func invalidIntent(fail failure.InvalidIntent) Error {
 		fail.Description,
 	)
 }
+
+func invalidAuthorizers(fail failure.InvalidAuthorizers) Error {
+	return convertError(
+		configuration.ErrorInvalidAuthorizers,
+		fail.Description,
+		withDetail("have_authorizers", fail.Have),
+		withDetail("want_authorizers", fail.Want),
+	)
+}
+
+func invalidPayer(fail failure.InvalidPayer) Error {
+	return convertError(
+		configuration.ErrorInvalidPayer,
+		fail.Description,
+		withAddress("have_payer", fail.Have),
+		withAddress("want_payer", fail.Want),
+	)
+}
+
+func invalidProposer(fail failure.InvalidProposer) Error {
+	return convertError(
+		configuration.ErrorInvalidProposer,
+		fail.Description,
+		withAddress("have_proposer", fail.Have),
+		withAddress("want_proposer", fail.Want),
+	)
+}
+
+func invalidScript(fail failure.InvalidScript) Error {
+	return convertError(
+		configuration.ErrorInvalidScript,
+		fail.Description,
+	)
+}
+
+func invalidArguments(fail failure.InvalidArguments) Error {
+	return convertError(
+		configuration.ErrorInvalidArguments,
+		fail.Description,
+		withDetail("have_arguments", fail.Have),
+		withDetail("want_arguments", fail.Want),
+	)
+}
+
+func invalidAmount(fail failure.InvalidAmount) Error {
+	return convertError(
+		configuration.ErrorInvalidAmount,
+		fail.Description,
+		withDetail("raw_amount", fail.RawAmount),
+	)
+}
+
+func invalidReceiver(fail failure.InvalidReceiver) Error {
+	return convertError(
+		configuration.ErrorInvalidReceiver,
+		fail.Description,
+		withDetail("raw_receiver", fail.RawReceiver),
+	)
+}
+
+func invalidSignature(fail failure.InvalidSignature) Error {
+	return convertError(
+		configuration.ErrorInvalidSignature,
+		fail.Description,
```

Comment:
Same here, maybe include the signature as hex?

Reply:
I have added it with `failure.WithString()` in `ParseTransaction()` function.

Since I used the `InvalidSignature` as a sentinel error for wrong signer and wrong number of signatures, existence of `signature` isn't common in all those scenarios.

It could potentially be split into two errors `InvalidSignatures` for wrong number, and `InvalidSigner` for when the signer is wrong.